### DISCUSSION TITLE
mixed mode search in DIA-NN

### DISCUSSION THREAD

**Discussion by weiclav** [vdemichev/DiaNN#144] [Q&A] [answered]
Hi there,

I would like to know what is the best way to set up a mixed search mode in DIA-NN, i.e. to use library from e.g. gas phase fractionation (GPF) experiment and to use also provided FASTA files to check for the peptides not covered in the library.

I am currently experimenting with addition of GPF data to the individual samples measurements with MBR enabled, but the improvements are quite small compared to the library-free search mode (it may be potentially related to the fact the data comes from TIMS).

Thank you in advance!

Best,
David

**Comment by vdemichev** [comment, marked as answer]
Hi David,

What one can do is analyse your DIA experiment and your GPF runs in a single go in library-free mode - to create a spectral library. Then reanalyse the DIA experiment with this library. Filter the output based on Q.Value, Lib.Q.Value and Lib.PG.Q.Value. The advantage of this over MBR would be that you can use the 'Unrelated runs' option during the library-creation stage only, allowing, for example, DIA-NN to use different scan window settings for different runs. But if the optimal scan window/mass accuracies are the same for DIA and GPF runs, yes, can just use MBR.

Yes, improvement in comparison to pure library-free might be small, because library-free in the latest DIA-NN is exceptionally good.

Alternatively, can create a library from your DIA data, and then just load it along with the GPF library, using --lib command twice. But the way to do it as described above is I think preferable.

Best,
Vadim

**Reply by weiclav** [reply]
Hi Vadim,

thanks a lot for the response!

So if I got you right regarding the 1st approach mentioned - no MBR is actually needed neither during the library preparation stage and also during the samples DIA data, since all peptides should be already part of the library, right?

And I kind of like the last mentioned approach as well since it can be done separately, so will give it a try as well. Any reason for not using this - e.g. due to the way the DIA and GPF data are combined during the library creation compared to using them separarely?

Thanks again!

Best,
David

**Reply by vdemichev** [reply]
Hi David,

Sorry for the very late reply, I somehow missed the github notification...

Yes, no MBR while generating a library. MBR only for quantitative analyses.
No idea which approach is better. I would think using --lib twice is more likely to be better, but I don't have any evidence for this.

Best,
Vadim

**Reply by weiclav** [reply]
Hi Vadim,

sorry for later follow up here.

So, to be 100% sure on when to use MBR and how to filter the generated matrix (I got a bit uncertain based on the discussions here on github, sorry; all is mentioned with intention to use the main report file, 1% FDR):
1) when you are performing library free search and you would like to get quantitatively valid data, MBR should be switched on; the main output should be filtered based on Lib.Q.Value and Lib.PG.Q.Value

2) if you want to only build library from e.g. all sample analyses and or GPF analyses of pooled sample, MBR should be switched off, only "Generate spectral library" checkbox should be checked; no filtering applicable, just use the generated library files - speclib format in case you will use just single library, tsv format in case of multiple libraries utilization (e.g. one based on GPF analyses and the other one based on samples analyses as mentioned as one possible approach above)

3) when using one or more project specific libraries (generated as indicated in the option 2), MBR should be switched off even for generation of data to be quantitatively evaluated later on (since the libraries are highly specific for the given set of samples); Global.Q.Value and Global.PG.Q.Value should be used to filter the main output

Thank you very much for your response.
David

**Reply by vdemichev** [reply]
Hi David, 

A note: whenever you use MBR, you always automatically get also a non-MBR report ('first-pass' files). So MBR is just something extra DIA-NN does. So if in doubt, just run with MBR and then compare the performance with/without, based on the respective reports.

1. Yes, MBR is strongly recommended for quantitative library-free searches. In this case it offers a tremedous improvement. Yes, Lib.Q.Value, Lib.PG.Q.Value, also Q.Value (if set to >1% in the GUI) and (optionally) PG.Q.Value (run-specific protein q-value, unique to DIA-NN, see the FAQ on FDR filtering).
2. I suggest to always use .tsv for any 'serious' analyses. It's better for reproducibility.
3. Yes, for very-specific library MBR usually need to be switched off. For GFP - that's almost certain. For DDA-based project-specific libs I have seen substantial improvement with MBR in some cases.

Best,
Vadim

**Reply by weiclav** [reply]
Thank you  very much for the update!

Take care,
David

**Reply by weiclav** [reply]
Hi Vadim,

sorry to bug you with this one once more. I am still uncertain with the main report filtration when the DIA sample runs and GPF runs libraries are used and MBR is not checked or checked but first pass report is used, i.e. 3rd scenario mentioned [here](https://github.com/vdemichev/DiaNN/discussions/144#discussioncomment-1313399).

Should one use Global.x or Lib.x q values instead in a situation where you have project specific libraries so big part of the library is being reported? You have mentioned in filtering on these three levels before in your first response here: Q.Value, Lib.Q.Value and Lib.PG.Q.Value; is this correct?

Thank you again!

Best,
David

**Reply by vdemichev** [reply]
Hi David,

If you don't use MBR, use q-values marked as 'Global' for global filtering. If you use MBR, use q-values marked as 'Lib' for global filtering.

Best,
Vadim

**Reply by weiclav** [reply]
Hi Vadim,

thanks for the confirmation!

Best regards,
David
